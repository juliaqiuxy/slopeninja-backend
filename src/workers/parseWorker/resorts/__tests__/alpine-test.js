import fs from 'fs';
import {
  parseAlpineSnow,
  parseAlpineLiftCounts,
  parseAlpineLifts,
  parseAlpineTrails,
  parseAlpineTrailCounts,
} from '../alpine';
import { createJSONParser } from '../../parserFactory';

test('fetches Alpine snow data correctly', async () => {
  const jsonText = fs.readFileSync(`${__dirname}/__fixtures__/alpine-lifts.json.input`);
  const resortData = await createJSONParser('snow', parseAlpineSnow)(jsonText);

  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: 'cloudy',
      temperature: 43,
      baseCondition: 'Corn snow',
      newSnow: 0,
      snowDepthBase: 19,
      snowDepthSummit: 27,
    },
  });
});

test('fetches all null for nonexisting snow values', async () => {
  const resortData = await createJSONParser('snow', parseAlpineSnow)('{}');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches Alpine lifts data correctly', async () => {
  const jsonText = fs.readFileSync(`${__dirname}/__fixtures__/alpine-lifts.json.input`);
  const resortData = await createJSONParser('liftCounts', parseAlpineLiftCounts)(jsonText);

  expect(resortData).toEqual({
    liftCounts: {
      total: 12,
      open: 0,
    },
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createJSONParser('liftCounts', parseAlpineLiftCounts)('{}');
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Alpine trails data correctly', async () => {
  const jsonText = fs.readFileSync(`${__dirname}/__fixtures__/alpine-lifts.json.input`);
  const resortData = await createJSONParser('trailCounts', parseAlpineTrailCounts)(jsonText);

  expect(resortData).toEqual({
    trailCounts: {
      total: 115,
      open: 0,
    },
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createJSONParser('trailCounts', parseAlpineTrailCounts)('{}');
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Alpine lift list correctly', async () => {
  const jsonText = fs.readFileSync(`${__dirname}/__fixtures__/alpine-lifts.json.input`);
  const resortData = await createJSONParser('lifts', parseAlpineLifts)(jsonText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createJSONParser('lifts', parseAlpineLifts)('{}');

  expect(resortData).toMatchObject({ lifts: [] });
});

test('fetches Alpine trail list correctly', async () => {
  const jsonText = fs.readFileSync(`${__dirname}/__fixtures__/alpine-lifts.json.input`);
  const resortData = await createJSONParser('trails', parseAlpineTrails)(jsonText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createJSONParser('trails', parseAlpineTrails)('{}');
  expect(resortData).toMatchObject({ trails: [] });
});
