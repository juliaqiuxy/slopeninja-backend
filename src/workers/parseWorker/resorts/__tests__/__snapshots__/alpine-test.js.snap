// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Alpine lift list correctly 1`] = `
Object {
  "lifts": Array [
    Object {
      "category": null,
      "name": "Kangaroo",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Lakeview",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Big Carpet",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Meadow",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Roundhouse",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Scott",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Sherwood Express",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Subway",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Alpine Bowl Chair",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Summit Express",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Yellow",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Treeline Cirque",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Little Carpet",
      "status": "closed",
    },
  ],
}
`;

exports[`fetches Alpine trail list correctly 1`] = `
Object {
  "trails": Array [
    Object {
      "category": "Kangaroo",
      "level": "intermediate",
      "name": "Kangaroo Main Run",
      "status": "closed",
    },
    Object {
      "category": "Kangaroo",
      "level": "intermediate",
      "name": "Nick's Run",
      "status": "closed",
    },
    Object {
      "category": "Lakeview",
      "level": "intermediate",
      "name": "Lakeview to Weasel",
      "status": "closed",
    },
    Object {
      "category": "Lakeview",
      "level": "intermediate",
      "name": "Leisure Lane",
      "status": "closed",
    },
    Object {
      "category": "Lakeview",
      "level": "intermediate",
      "name": "Mountain View",
      "status": "closed",
    },
    Object {
      "category": "Lakeview",
      "level": "intermediate",
      "name": "Outer Limits",
      "status": "closed",
    },
    Object {
      "category": "Lakeview",
      "level": "intermediate",
      "name": "Scotty's Beam",
      "status": "closed",
    },
    Object {
      "category": "Lakeview",
      "level": "intermediate",
      "name": "Shooting Star",
      "status": "closed",
    },
    Object {
      "category": "Lakeview",
      "level": "intermediate",
      "name": "Twilight Zone",
      "status": "closed",
    },
    Object {
      "category": "Meadow",
      "level": "beginner",
      "name": "Meadow Beginner Area",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Banana Chute",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Boomerang",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Charity",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Dance Floor",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "East Runout",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "advanced",
      "name": "Gunner's Knob",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "advanced",
      "name": "Hot Wheels Gully",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Loop Road",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Rock Garden",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Sandy's Corner",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "advanced",
      "name": "Sympathy Face",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Upper Weasel One",
      "status": "closed",
    },
    Object {
      "category": "Roundhouse",
      "level": "intermediate",
      "name": "Werner's Schuss",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "intermediate",
      "name": "Bobby's Run",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "advanced",
      "name": "Chute That Seldom Slides",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "advanced",
      "name": "Gentian Gully",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "advanced",
      "name": "Lower 40 Face",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "advanced",
      "name": "Lower 40 Gully",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "advanced",
      "name": "Promised Land",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "advanced",
      "name": "Scott Chute",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "intermediate",
      "name": "Scott Meadow",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "intermediate",
      "name": "Scott Ridge",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "intermediate",
      "name": "Standard Run",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "advanced",
      "name": "Subway Cirque",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "intermediate",
      "name": "Summer Road",
      "status": "closed",
    },
    Object {
      "category": "Scott",
      "level": "intermediate",
      "name": "Winter Road",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "advanced",
      "name": "East Gully",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "intermediate",
      "name": "Maid Marian",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "intermediate",
      "name": "Nottingham's Notch",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "intermediate",
      "name": "Ray's Rut",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "intermediate",
      "name": "Reily's Run",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "intermediate",
      "name": "Robin Hood",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "advanced",
      "name": "Sherwood Face",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "intermediate",
      "name": "Sherwood Run",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "advanced",
      "name": "South Face",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "advanced",
      "name": "South Face Access",
      "status": "closed",
    },
    Object {
      "category": "Sherwood",
      "level": "advanced",
      "name": "West Gully",
      "status": "closed",
    },
    Object {
      "category": "Subway",
      "level": "beginner",
      "name": "Subway Beginner Area",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "intermediate",
      "name": "Alpine Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Beaver Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Bernie's Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Big Bend Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Big Bend Keyhole",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "CB Chute",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "CK Bench",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "D5",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "D6",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "D7",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "D8",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Estelle Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Grouse Rock",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Grouse Saddle",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "High Traverse",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "High Yellow Face",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "High Yellow Gully",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Idiot's Delight",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "intermediate",
      "name": "Kangaroo Ridge",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Keyhole",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Keyhole Slopes",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Lower Beaver Entrance",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Lower Saddle",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "North Peril",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Palisades",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Peril Ridge",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Pete's Peril",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Poma Rocks",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Pond Slope",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Sherwood Creek",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "South Beaver",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "S.P. Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Sun Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "intermediate",
      "name": "Sunspot",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "intermediate",
      "name": "Terry's Return",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "The Buttress",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "The Face",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "The Face Cliffs",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Three Sisters",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Top Alpine Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Upper Saddle",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Waterfall",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "intermediate",
      "name": "West Runout",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "intermediate",
      "name": "Wolverine",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Wolverine Bowl",
      "status": "closed",
    },
    Object {
      "category": "Summit & Alpine Bowl",
      "level": "advanced",
      "name": "Wolverine Saddle",
      "status": "closed",
    },
    Object {
      "category": "Terrain Parks",
      "level": null,
      "name": "Burton Riglet Park",
      "status": "closed",
    },
    Object {
      "category": "Terrain Parks",
      "level": null,
      "name": "Kangaroo Ridge Park",
      "status": "closed",
    },
    Object {
      "category": "Terrain Parks",
      "level": null,
      "name": "Loop Road Family Cross",
      "status": "closed",
    },
    Object {
      "category": "Terrain Parks",
      "level": null,
      "name": "Start Park",
      "status": "closed",
    },
    Object {
      "category": "Terrain Parks",
      "level": null,
      "name": "Tiegel Terrain Park",
      "status": "closed",
    },
    Object {
      "category": "Yellow",
      "level": "intermediate",
      "name": "Fall Line",
      "status": "closed",
    },
    Object {
      "category": "Yellow",
      "level": "intermediate",
      "name": "Ladies' Slalom",
      "status": "closed",
    },
    Object {
      "category": "Yellow",
      "level": "intermediate",
      "name": "Yellow Trail",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "advanced",
      "name": "Art's Knob",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "advanced",
      "name": "Counterweight Gully",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "intermediate",
      "name": "East Creek",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "intermediate",
      "name": "Easy Return Road",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "advanced",
      "name": "Expert Shortcut",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "advanced",
      "name": "Hidden Knolls",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "advanced",
      "name": "Our Father",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "intermediate",
      "name": "Return Road",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "advanced",
      "name": "Sherwood Cliffs",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "intermediate",
      "name": "Skateboard Alley",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "intermediate",
      "name": "Tiegel ",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "intermediate",
      "name": "To Lakeview",
      "status": "closed",
    },
    Object {
      "category": "Treeline Cirque",
      "level": "intermediate",
      "name": "Weasel Run",
      "status": "closed",
    },
  ],
}
`;
